src/Timezone/UserBasedTimezoneDetector.php

```diff
@@ -36,15 +36,29 @@ public function __construct(TokenStorageInterface $securityContext)
     public function getTimezone()
     {
         if (!$token = $this->securityContext->getToken()) {
-            return;
+            return null;
         }
 
         if (!$user = $token->getUser()) {
-            return;
+            return null;
         }
 
+        if ($user instanceof TimezoneAwareInterface) {
+            return $user->getTimezone();
+        }
+
+        // NEXT_MAJOR: Remove this check when `User` implement `TimezoneAwareInterface`
```

Comment:
And we should say to remove the 
```
**DEPRECATED**
Relying on ``Sonata\UserBundle\Model\User`` is deprecated since 2.x in favor of
explicit implementation of ``Sonata\IntlBundle\Timezone\TimezoneAwareInterface``.
```
section